cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -432,6 +436,24 @@ runTxBuildRaw (AnyCardanoEra era)
         in firstExceptT ShelleyTxCmdWriteFileError . newExceptT $
              writeTxFileTextEnvelopeCddl fpath noWitTx
 
+getSbe :: Monad m => CardanoEraStyle era -> ExceptT ShelleyTxCmdError m (Api.ShelleyBasedEra era)
```

Comment:
I adopted the changes in your commit.

I have also swapped the arguments of the `eraCast` function because putting the `era` first is more natural and prevents the need to introduce lambdas or flip arguments.

Reply:
I haven't moved any functions into `Cardano.CLI.Helpers` because the noted functions aren't exact duplicates of each other due to having different types.

Reply:
Incidentally, this is one of the reasons why using sum types for error types isn't good practise.

Every error type should only have one constructor.

https://www.parsonsmatt.org/2018/11/03/trouble_with_typed_errors.html

Reply:
I see your point however I think because the only place we pattern match on `ShelleyQueryCmdError` and `ShelleyTxCmdError` are in the error rendering functions this is ok. These errors are for diagnoses, they don't decide between code branches so we don't end up in the situation where we have to pattern match with  `_ -> error "impossible"` which I believe is the main gripe (and I agree, this is gross).